triqs/gfs/block/block_gf.hxx

```diff
@@ -209,6 +212,21 @@ namespace triqs {
       /// Construct from the vector of names
       block_gf(block_names_t b) : _block_names(std::move(b)), _glist(_block_names.size()) {}
 
+      // Create Block Green function from Mesh and gf_struct
+      block_gf(gf_mesh<Var> const &m, triqs::hilbert_space::gf_struct_t const &gf_struct) {
+
+        for (auto const & [ bname, idx_lst ] : gf_struct) {
+          auto bl_size = idx_lst.size();
+          _block_names.push_back(bname);
+          std::vector<std::string> idx_str_lst(bl_size);
+          std::transform(idx_lst.cbegin(), idx_lst.cend(), idx_str_lst.begin(), [](auto &arg) { return std::to_string(arg); });
+          if constexpr (Target::rank == 0)
+            _glist.emplace_back(m, make_shape(bl_size, bl_size));
+          else
+            _glist.emplace_back(m, make_shape(bl_size, bl_size), std::vector<std::vector<std::string>>{Target::rank, idx_str_lst});
```

Comment:
In fact, this will call the (int i, T val) constructor of vector, which inserts the same component i times. In the updated version I will use ( .. ) though instead of { .. } as the confusion with the initializer_list constructor is dangerous.

Reply:
I guess you are right.
C++ is extremely confusing and is getting even more so with every new version...